logic/Rule.java

```diff
@@ -241,41 +235,12 @@ public Rule rule() {
             return rule;
         }
 
-        public Conjunction conjunction() {
-            return rule().when();
-        }
-
-        public Set<Concludable> concludablesTriggeringRules(ConceptManager conceptMgr, LogicManager logicMgr) {
-            if (concludablesTriggeringRules == null) { // only acquire lock if required
-                synchronized (this) { // only compute concludables once
-                    if (concludablesTriggeringRules == null) {
-                        concludablesTriggeringRules = iterate(Concludable.create(rule.when()))
-                                .filter(c -> c.getApplicableRules(conceptMgr, logicMgr).hasNext()).toSet();
-                    }
-                }
-            }
-            return concludablesTriggeringRules;
-        }
-
-        public Set<Concludable> negatedConcludablesTriggeringRules(ConceptManager conceptMgr, LogicManager logicMgr) {
-            synchronized (this) { // can be more contentious as only used for validation
-                if (negatedConcludablesTriggeringRules == null) {
-                    negatedConcludablesTriggeringRules = concludables(rule.when().negations())
-                            .filter(c -> c.getApplicableRules(conceptMgr, logicMgr).hasNext()).toSet();
-                }
-            }
-            return negatedConcludablesTriggeringRules;
+        public ResolvableConjunction conjunction() {
+            return conjunction;
         }
 
-        private FunctionalIterator<Concludable> concludables(List<Negation> negations) {
-            return iterate(negations)
-                    .flatMap(neg -> {
-                        assert neg.disjunction().conjunctions().size() == 1;
-                        return iterate(neg.disjunction().conjunctions());
-                    }).flatMap(conjunction -> {
-                        assert conjunction.negations().isEmpty();
-                        return iterate(Concludable.create(conjunction));
-                    });
+        public Conjunction pattern() {
+            return rule().when();
         }
```

Comment:
Rename the method returning the pattern.Conjunction to 'pattern'
Use the `conjunction` name for the one returning a ResolvableConjunction.